Fix DOI/ID message when deaccession a Dataset

jp-tosca opened the pull request:
**What this PR does / why we need it**:
It was reported on #10309 that this parameter was not substituted by the value on the success message.

**Which issue(s) this PR closes**:
Closes #10309

**Suggestions on how to test this**:

1. Create a Dataset version
2. Deaccession it with `http://localhost:8080/api/datasets/:persistentId/versions/:versionID/deaccession?key=API_KEY&persistentId=doi:10.5072/FK2/RGFWNO`
3. the new message should show as `"message": "Dataset 10.5072/FK2/RGFWNO deaccessioned for version 9"`
4. Check the database id still works `http://localhost:8080/api/datasets/6/versions/10/deaccession?key=API_KEY`
5. message should show as `"message": "Dataset 6 deaccessioned for version 10"`

**Is there a release notes update needed for this change?**:
No

**Additional documentation**:
No

pdurbin commented on src/main/java/edu/harvard/iq/dataverse/api/Datasets.java:
```diff
@@ -4791,7 +4791,10 @@ public Response deaccessionDataset(@Context ContainerRequestContext crc, @PathPa
-                return ok("Dataset " + datasetId + " deaccessioned for version " + versionId);
+                
+                return ok("Dataset " + 
+                        (":persistentId".equals(datasetId) ? datasetVersion.getDataset().getAuthority() + "/" + datasetVersion.getDataset().getIdentifier() : datasetId) + 
```
Won't this be missing "doi:"? What if it's "hdl:" instead?

Can we please have a test?

pdurbin reviewed (commented):
I'm pretty sure this won't show the full PID.

jp-tosca commented:
> I'm pretty sure this won't show the full PID.

It won't, I removed it. Previously I was using the "protocol" would that be the right thing to use to prepend? @pdurbin

jp-tosca commented:
Thanks @qqmyers! I updated this PR to use ***asString*** from the GlobalID.

I also added to the tests to check when it is deleted by a specified ID to be contained on the message and a function to delete by global. @pdurbin

pdurbin commented on src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java:
```diff
@@ -3858,6 +3858,19 @@ static Response deaccessionDataset(Integer datasetId, String version, String dea
+                .body(jsonString)
+                .post("/api/datasets/:persistentId/versions/" + version + "/deaccession?persistentId=" + globalId);
+    }
+
```
I guess it's ok to add a whole new method but if you want to pass a PID (I would suggest this term over GlobalID), you could use this pattern you'll see elsewhere in UtilIT:

```
String idInPath = datasetIdOrPersistentId; // Assume it's a number.
String optionalQueryParam = ""; // If idOrPersistentId is a number we'll just put it in the path.
if (!NumberUtils.isCreatable(datasetIdOrPersistentId)) {
    idInPath = ":persistentId";
    optionalQueryParam = "?persistentId=" + datasetIdOrPersistentId;
}
```

jp-tosca replied:
I made some changes, please let me know if this is better.

pdurbin replied:
It's better, but did you find a datasetIdOrPersistentId example in UtilIT? Do you see what I mean, how we can reduce code duplication?

jp-tosca replied:
I will change that in a bit 😅

jp-tosca replied:
Ok, I made some changes, let me know what you think. 😄

pdurbin reviewed (commented):
This is coming along. I'm adding a suggestion for how to avoid adding an entire new method to UtilIT.

qqmyers reviewed (approved):
Looks good. Tests pass. Looks like all of the prior review comments have been addressed.
